src/PowerPing/Ping.cs

```diff
@@ -457,85 +457,50 @@ private PingResults SendICMP(PingAttributes attrs)
                     if (Display.ShowOutput) {
                         PowerPing.Display.Error("General transmit error");
                     }
-                    Results.SaveResponseTime(-1);
-                    try { Results.Lost++; }
-                    catch (OverflowException) { Results.HasOverflowed = true; }
+                    results.SaveResponseTime(-1);
+                    try { results.Lost++; }
+                    catch (OverflowException) { results.HasOverflowed = true; }
 
                 } catch (SocketException) {
 
                     PowerPing.Display.Timeout(index);
 		            if (attrs.BeepLevel == 1) {
                         try { Console.Beep(); }
-                        catch (Exception) { Results.HasOverflowed = true; }
+                        catch (Exception) { results.HasOverflowed = true; }
                     }
-                    Results.SaveResponseTime(-1);
-                    try { Results.Lost++; }
-                    catch (OverflowException) { Results.HasOverflowed = true; }
+                    results.SaveResponseTime(-1);
+                    try { results.Lost++; }
+                    catch (OverflowException) { results.HasOverflowed = true; }
+
+                } catch (OperationCanceledException) {
+
+                    break;
```

Comment:
Feel like we could specify that this break is for when user cancels the operation but I suppose the variables around tell that story so it's kind of a moot point.